pkg/loki/loki.go

```diff
@@ -546,28 +553,38 @@ func (t *Loki) setupModuleManager() error {
 		IngesterQuerier:          {Ring},
 		IndexGatewayRing:         {RuntimeConfig, Server, MemberlistKV},
 		All:                      {QueryScheduler, QueryFrontend, Querier, Ingester, Distributor, Ruler, Compactor},
-		Read:                     {QueryScheduler, QueryFrontend, Querier, Ruler, Compactor, IndexGateway},
+		Read:                     {QueryFrontend, Querier},
 		Write:                    {Ingester, Distributor},
+		Backend:                  {QueryScheduler, Ruler, Compactor, IndexGateway},
```

Comment:
Curious why the ruler is in this target... can you expand on that pls?

Reply:
Two reasons, 1) because that's what mimir did, 2) because we want to be able to run the Read target as a deployment that can easily be killied, OOMed, scaled up or down. How does the Ruler respond to unexpected interruptions? I do see we run it as a deployment, how problematic will it be if a ruler gets interrupted because a Querier is OOM killed?

Reply:
Gotcha, thanks for the context.
The ruler can respond just fine to interruptions, but it's better that they're a statefulset so that rules can be ~consistently sharded